src/modules/keyboardmanager/common/Helpers.cpp

```diff
@@ -208,30 +208,44 @@ namespace KeyboardManagerHelper
     // Function to return the executable name of the application in focus
     std::wstring GetCurrentApplication(bool keepPath)
     {
-        HWND current_window_handle = GetFocusWindowHandle();
-        DWORD process_id;
-        DWORD nSize = MAX_PATH;
-        WCHAR buffer[MAX_PATH] = { 0 };
-
-        // Get process ID of the focus window
-        DWORD thread_id = GetWindowThreadProcessId(current_window_handle, &process_id);
-        HANDLE hProc = OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_VM_READ, FALSE, process_id);
-
-        // Get full path of the executable
-        bool res = QueryFullProcessImageName(hProc, 0, buffer, &nSize);
+        HWND current_window_handle = GetForegroundWindow();
         std::wstring process_name;
-        CloseHandle(hProc);
 
-        process_name = buffer;
-        if (res)
+        if (current_window_handle != nullptr)
         {
-            PathStripPath(buffer);
+            std::wstring process_path = get_process_path(current_window_handle);
+            process_name = process_path;
+            
+            // Get process name from path
+            PathStripPath(&process_path[0]);
 
+            // Remove elements after null character
+            process_path.erase(std::find(process_path.begin(), process_path.end(), L'\0'), process_path.end());
+
+            // If the UWP app is in full-screen, then using GetForegroundWindow approach might fail
+            if (process_path == L"ApplicationFrameHost.exe")
+            {
+                HWND fullscreen_window_handle = GetFullscreenUWPWindowHandle();
+                if (fullscreen_window_handle != nullptr)
+                {
+                    process_path = get_process_path(fullscreen_window_handle);
+                    process_name = process_path;
+
+                    // Get process name from path
+                    PathStripPath(&process_path[0]);
+
+                    // Remove elements after null character
+                    process_path.erase(std::find(process_path.begin(), process_path.end(), L'\0'), process_path.end());
+                }
+            }
+
+            // If keepPath is false, then return only the name of the process
             if (!keepPath)

```

Comment:
As of now we only need the process name, however in the future when we add support for detecting installed applications (like what PT Run does), we would be checking the whole process path rather than just the process name. An example of what we would be checking would be `chrome.exe` versus `C:\Program Files\Chrome\chrome.exe`.

Reply:
Just curious, why would we need the path even then? Wouldn't just the name be sufficient? It needs to behave like `task manager` or `attach to process` from VS, and they just have the `process name`.

Reply:
It would be a safety thing. For example if a user had an app-specific remap for PowerToys, then we would only want it to run from the installed version and not from another path (for example a debug build).